File: Nian_Karl/QL/src/ast/expression/unary/Unary.java
package ast.expression.unary;

import ast.expression.Expression;

public abstract class Unary extends Expression {

	protected final Expression expr;
	
	public Unary(Expression expr) {
		this.expr = expr;
	}
	
	public Expression getExpression() {
		return expr;
	}

}

File: FelipezConde/QuestionnaireLanguage/AST/Nodes/Expressions/Unary/Unary.cs
using Expressions = AST.Nodes.Interfaces;

namespace AST.Nodes.Expressions.Unary
{
    public abstract class Unary : Expression
    {
        private readonly Expression child;

        protected Unary(Expression child, Representation.PositionInText pos)
            :base(pos)
        { this.child = child; }

        public Expression GetChildExpression()
        { return child; }

        public abstract override T Accept<T>(ASTVisitors.IVisitor<T> visitor);
    }
}
